### PR TITLE
Added User & Conversation Avatar & minor UI improvements

### PR DESCRIPTION
**Pull request by abhishekbadoni-bigstep**
Worked on
- showing user initials if user does not have profile image.
- improved minor padding margin issues.
- Removed "No conversation selected" in case of live stream
- added host keyword for message done by host.

### CODE REVIEW

File: src/actions/messageActions.js [hunk 1]
```diff
@@ -235,4 +241,54 @@ export const registerConversationEventHandlers = (conversation) => {
...
+      type: START_WATCHING_PROGRESS,
+      payload: {}
+    });
+    return conversation.startWatching((err, response) => {
+      if (err) {
+        dispatch({
+          type: START_WATCHING_FAIL,
+          payload: err
+        });
+      }
```
Comment (balramgoyal) [inline src/actions/messageActions.js:259]: @abhishekbadoni-bigstep Use `return` here.

File: src/actions/messageActions.js [hunk 2]
```diff
@@ -235,4 +241,54 @@ export const registerConversationEventHandlers = (conversation) => {
...
+      type: STOP_WATCHING_PROGRESS,
+      payload: {}
+    });
+    return conversation.stopWatching((err, response) => {
+      if (err) {
+        dispatch({
+          type: STOP_WATCHING_FAIL,
+          payload: err
+        });
+      }
```
Comment (balramgoyal) [inline src/actions/messageActions.js:285]: @abhishekbadoni-bigstep Please use `return` here.

File: src/components/ConversationIcon.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  reactAvatarProps: PropTypes.object
+}
+
+ConversationIcon.defaultProps = {
+  extraContent: null,
+  extraContentArguments: [],
+  reactAvatarProps: {}
+};
+
+export { ConversationIcon }
```
Comment (balramgoyal) [inline src/components/ConversationIcon.js]: @abhishekbadoni-bigstep I believe that this component should be general ans its name should be like `Avatar`. Moreover this should not have `conversation prop`, this should only have `initials` & `src`.

File: src/components/ConversationIcon.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+          !hasProfileImage &&
+          <div title={title} className= {className} style={{ backgroundImage: `url(${profileImageUrl})` }}>
+            <Avatar name={title} {...finalReactAvatarProps} />
+            {extraContent && extraContent(...extraContentArguments)}          
+          </div>
+        }
+        {
+          hasProfileImage &&
+          <div title={title} className= {className} style={{ backgroundImage: `url(${profileImageUrl})` }}>
+            {extraContent && extraContent(...extraContentArguments)}
```
Comment (balramgoyal) [inline src/components/ConversationIcon.js]: I think that this should also be possible by `Avatar` component. Please check it.

File: src/components/ConversationIcon.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const {
+      conversation,
+      className,
+      extraContent,
```
Comment (balramgoyal) [inline src/components/ConversationIcon.js]: This is not needed as `this.props.children` as the children components. You can use this.

File: src/components/ConversationList.js [hunk 6]
```diff
@@ -114,19 +114,14 @@ class ConversationList extends PureComponent {
...
-    }
 
     list = list.map(conversation => modifyConversation(conversation));
 
     return (
       <div id="ch_recent_window" className="ch-recent-window">
         <Header 
-          profileImageUrl={headerImage}
+          profileImageUrl={user && user.profileImageUrl}
+          profileImageAlt={user && user.displayName}
```
Comment (balramgoyal) [inline src/components/ConversationList.js]: `profileImageAlt` should be renamed as `initials`.

File: src/components/ConversationWindow.js [hunk 7]
```diff
@@ -586,6 +593,7 @@ class ConversationWindow extends PureComponent {
                 return <Message 
                     key={message.id} 
                     message={message} 
+                    isMessageByAdmin={ conversationAdmins.includes(message.ownerId) }
```
Comment (balramgoyal) [inline src/components/ConversationWindow.js]: @abhishekbadoni-bigstep we also can make a function `isSentByAdmin` and we can use this function in this component.

File: src/components/ConversationWindow.js [hunk 8]
```diff
@@ -663,7 +671,8 @@ ConversationWindow.propTypes = {
 ConversationWindow.defaultProps = {
   Message: MessageSimple,
   showHeader: true,
-  showComposerActions: true
+  showComposerActions: true,
+  noRecordFoundError: LANGUAGE_PHRASES.NO_CONVERSATION_SELECTED
```
Comment (balramgoyal) [inline src/components/ConversationWindow.js]: Please rename this to `noConversationFoundMessage` as `noRecordFoundError` is making sense for no conversation found as well as no messages found.

File: src/components/Header.js [hunk 9]
```diff
@@ -28,11 +30,15 @@ class Header extends PureComponent {
...
     const { showDropDownList } = this.state;
+    const hasProfileImage = profileImageUrl && profileImageUrl.trim();
 
     return (
     <div id="ch_header" className="ch-header">
       { showArrowBack && <i className="material-icons arrow-back" onClick={onBack}>arrow_back</i>}
-      { profileImageUrl && <div className="ch-header-image" style={{backgroundImage:`url(${profileImageUrl})`}}></div>}
+
+      { hasProfileImage && <div className="ch-header-image" title={profileImageAlt} style={{backgroundImage:`url(${profileImageUrl})`}}></div>}
+      { !hasProfileImage && profileImageAlt && <Avatar name={profileImageAlt} className="ch-header-image" maxInitials={2} size={40} round={true} /> }
```
Comment (balramgoyal) [inline src/components/Header.js]: @abhishekbadoni-bigstep you can write this condition in ternary form.

File: src/components/MessageLivestream.js [hunk 10]
```diff
@@ -87,12 +87,16 @@ class MessageLivestream extends Component {
...
-						<div title={message.owner.displayName} className="ch-message-owner-avatar" style={{backgroundImage:`url(${ownerProfileImageUrl})`}}></div>
+					<div className={`ch-msg-container ch-msg-container-livestream`}>
+
+						<UserIcon user={message.owner} className="ch-message-owner-avatar"></UserIcon>
+
 						<div className={`ch-msg-content ch-msg-content__livestream`}>
 								{ message.body && 
 									<div className={`ch-text-message`}>
-										<div className="ch-message-owner-name">{message.owner.displayName}</div>
+										<div className="ch-message-owner-name">{message.owner.displayName}
```
Comment (balramgoyal) [inline src/components/MessageLivestream.js:97]: @abhishekbadoni-bigstep are not we using `avatar` as we were supposed to use avatar here also.

File: src/components/UserIcon.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+  reactAvatarProps: PropTypes.object
+}
+
+UserIcon.defaultProps = {
+  extraContent: null,
+  extraContentArguments: [],
+  reactAvatarProps: {}
+};
+
+export { UserIcon }
```
Comment (balramgoyal) [inline src/components/UserIcon.js]: @abhishekbadoni-bigstep we can use the upper component `AVATAR` here. do we need a separate component here?

File: src/actions/messageActions.js (hunk 1 above)
Comment (abhishekbadoni-bigstep) [reply src/actions/messageActions.js:259]: Done!

File: src/actions/messageActions.js (hunk 2 above)
Comment (abhishekbadoni-bigstep) [reply src/actions/messageActions.js:285]: Done!

File: src/components/ConversationIcon.js (hunk 3 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationIcon.js]: Done!

File: src/components/ConversationIcon.js (hunk 5 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationIcon.js]: Done!

File: src/components/ConversationList.js (hunk 6 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationList.js]: Done!

File: src/components/ConversationWindow.js (hunk 8 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationWindow.js]: Done!

File: src/components/Header.js (hunk 9 above)
Comment (abhishekbadoni-bigstep) [reply src/components/Header.js]: Done!

File: src/components/MessageLivestream.js (hunk 10 above)
Comment (abhishekbadoni-bigstep) [reply src/components/MessageLivestream.js:97]: We are using avatar here also.

File: src/components/ConversationIcon.js (hunk 4 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationIcon.js]: Done!

File: src/components/ConversationWindow.js (hunk 7 above)
Comment (abhishekbadoni-bigstep) [reply src/components/ConversationWindow.js]: Done!

File: src/components/UserIcon.js (hunk 11 above)
Comment (abhishekbadoni-bigstep) [reply src/components/UserIcon.js]: Done!

**Pull request merged**